packages/diracx-web-components/components/shared/FilterForm.tsx

```diff
@@ -116,34 +139,155 @@ export function FilterForm(props: FilterFormProps) {
             </Select>
           </FormControl>
 
-          <FormControl variant="outlined" fullWidth>
+          <FormControl variant="outlined">
             <InputLabel id="operator">Operator</InputLabel>
             <Select
               value={tempFilter.operator}
-              onChange={(e) => onChange("operator", e.target.value)}
+              onChange={(e) => {
+                onChange(
+                  ["in", "not in"].includes(e.target.value)
+                    ? "values"
+                    : "value",
+                  [] || "",
+                );
+                onChange(
+                  ["in", "not in"].includes(e.target.value)
+                    ? "value"
+                    : "values",
+                  undefined,
+                );
+                onChange("operator", e.target.value);
+              }}
               label="Operator"
               labelId="operator"
               data-testid="filter-form-select-operator"
               sx={{ minWidth: 120 }}
             >
-              <MenuItem value="eq">equals to</MenuItem>
-              <MenuItem value="neq">not equals to</MenuItem>
-              <MenuItem value="gt">is greater than</MenuItem>
-              <MenuItem value="lt">is lower than</MenuItem>
-              <MenuItem value="like">like</MenuItem>
+              {selectedColumn?.type != "DateTime" && (
+                <MenuItem value="eq">equals to</MenuItem>
+              )}
+              {selectedColumn?.type != "DateTime" && (
+                <MenuItem value="neq">not equals to</MenuItem>
+              )}
+
+              {selectedColumn?.type == "DateTime" && (
+                <MenuItem value="last">in the last</MenuItem>
+              )}
+
+              {typeof selectedColumn?.type != "object" && (
+                <MenuItem value="gt">is greater than</MenuItem>
+              )}
+              {typeof selectedColumn?.type != "object" && (
+                <MenuItem value="lt">is lower than</MenuItem>
+              )}
+              {(typeof selectedColumn?.type == "object" ||
+                selectedColumn?.type == "number") && (
+                <MenuItem value="in">is in</MenuItem>
+              )}
+              {(typeof selectedColumn?.type == "object" ||
+                selectedColumn?.type == "number") && (
+                <MenuItem value="not in">is not in</MenuItem>
+              )}
+              {selectedColumn?.type != "DateTime" && (
+                <MenuItem value="like">like</MenuItem>
+              )}
```

Comment:
I wonder if it would not be easier to manage on the long term and easier to read if we have a structure with each type of filter, such as:

```
DateTime:
- supported filters: last, gt, lt
- default filter: last

Number:
- supported filters: eq, neq, gt, lt, in, not in, like
- default filter: eq

String:
- supported filters: eq, neq, gt, lt, in, not in, like
- default filter: eq

Category:
- supported filters: eq, neq, gt, lt, in, not in, like
- default filter: eq
```

Any thought?